contracts/V1/StateChannelDiamondProxy/StateSnapshotFacet.sol

```diff
@@ -0,0 +1,230 @@
+import "../DataTypes.sol";
+import "./AStateChannelManagerProxy.sol";
+
+contract StateSnapshotFacet is StateChannelCommon {
+
+
+    function updateStateSnapshotWithDispute(
+        bytes32 channelId,
+        ForkMilestoneProof[] memory milestoneProofs,
+        StateSnapshot[] memory milestoneSnapshots,
+        DisputeProof memory disputeProof,
+        ExitChannelBlock[] memory exitChannelBlocks
+    ) external onlySelf {
+        // resolve genesis state snapshot source
+        // - if stateSnapshot(on-chain).forkCnt == disputeProof.forkCnt, then the genesis state snapshot is the on-chain stateSnapshot
+        // - otherwise, the dispute is validated and the genesis state snapshot is disputeProof.outputStateSnapshot
+        StateSnapshot memory genesisStateSnapshot = _resolveGenesisSnapshot(
+            channelId,
+            disputeProof
+        );
+
+        _updateStateSnapshot(
+            channelId,
+            milestoneProofs,
+            milestoneSnapshots,
+            exitChannelBlocks,
+            genesisStateSnapshot
+        );
+    }
+
+    function updateStateSnapshotWithoutDispute(
+        bytes32 channelId,
+        ForkMilestoneProof[] memory milestoneProofs,
+        StateSnapshot[] memory milestoneSnapshots,
+        ExitChannelBlock[] memory exitChannelBlocks
+    ) external onlySelf {
+        require(
+            getDisputeLength(channelId) == getSnapshotForkCnt(channelId),
+            "Dispute proof is required"
+        );
+        StateSnapshot memory genesisStateSnapshot = stateSnapshots[channelId];
+
+        _updateStateSnapshot(
+            channelId,
+            milestoneProofs,
+            milestoneSnapshots,
+            exitChannelBlocks,
+            genesisStateSnapshot
+        );
+    }
+
+    function _resolveGenesisSnapshot(
+        bytes32 channelId,
+        DisputeProof memory disputeProof
+    ) internal view returns (StateSnapshot memory) {
+        if (getDisputeLength(channelId) != getSnapshotForkCnt(channelId)) {
+            // dispute proof is required
+            Dispute memory dispute = disputeProof.dispute;
+            uint disputeTimestamp = disputeProof.timestamp;
+
+            require(
+                _isDisputeProofProvided(disputeTimestamp),
+                "Dispute proof is required"
+            );
+            require(
+                _isDisputeCommitmentValid(dispute, disputeTimestamp, channelId),
+                "Dispute proof is not valid"
+            );
+            require(
+                _isDisputeFinalized(disputeTimestamp),
+                "Dispute is not finalized"
+            );
+            require(
+                _isStateSnapshotValid(
+                    disputeProof.outputStateSnapshot,
+                    dispute
+                ),
+                "State snapshot is not valid"
+            );
+
+            return disputeProof.outputStateSnapshot;
+        } else {
+            return stateSnapshots[channelId];
```

Comment:
you can flip the condition in the 'if' and have the single return stateSnapshots[channelId] and instead of the 'else' you just have everything else normally - either way it's the same and not a big difference

Reply:
i also prefer early returns over if/else